src/Lecture4.hs

```diff
@@ -134,8 +142,22 @@ errors. We will simply return an optional result here.
 🕯 HINT: Use the 'readMaybe' function from the 'Text.Read' module.
 -}
 
+splitBy :: Char -> String -> [String]
+splitBy _ "" = []
+splitBy c s = case break (== c) s of
+  (pref, "") -> [pref]
+  (pref, _ : suff) -> pref : splitBy c suff
+
 parseRow :: String -> Maybe Row
-parseRow = error "TODO"
+parseRow = parseRow' . splitBy ','
+  where
+    parseRow' :: [String] -> Maybe Row
+    parseRow' [prd, trd, cost] =
+      do
+        trd' <- readMaybe trd
+        cost' <- readMaybe cost
+        pure Row {rowProduct = prd, rowTradeType = trd', rowCost = cost'}
+    parseRow' _ = Nothing
```

Comment:
I see that CI fails for this function:

* https://github.com/Kaelnor/exercises/actions/runs/4907476912/jobs/8762603554?pr=4

Specifically, the following tests are falling:

```haskell
it "Nothing on empty product"   $ parseRow ",Buy,10"             `shouldBe` Nothing
it "Nothing on negative number" $ parseRow "unknown,Buy,-5"      `shouldBe` Nothing
```

Some other tests for other functions fail as well but I believe this is mostly due to errors in this parsing function.

Reply:
Oops ! I forgot to run tests locally before pushing 🙄 I now check for both of these cases.